src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1932,24 +1932,26 @@ void PerformKeepAlive(object? state)
 
         #region Wait
 
-        public bool Wait(int timeout)
+        public async Task<bool> Wait(bool async, int timeout, CancellationToken cancellationToken = default)
```

Comment:
It is so cool to have just one unified implementation now :) I'm really impressed that we have a single system that behaves the same way across sync and async timeouts, which are so different in the end.

Reply:
It definitely is.